TWTValidation/JSON Validator/Parser/TWTJSONSchemaParser.m

```diff
@@ -212,40 +190,75 @@ - (TWTJSONSchemaNumberASTNode *)parseNumberSchema:(NSDictionary *)numberSchema
     if (node.minimum) {
         node.exclusiveMinimum = [self parseBooleanForKey:TWTJSONSchemaKeywordExclusiveMinimum schema:numberSchema valueIfNotPresent:NO];
     }
-
-    return node;
 }
 
 
-- (TWTJSONSchemaStringASTNode *)parseStringSchema:(NSDictionary *)stringSchema
+- (void)parseStringSchema:(NSDictionary *)stringSchema intoNode:(TWTJSONSchemaStringASTNode *)node
 {
-    TWTJSONSchemaStringASTNode *node = [[TWTJSONSchemaStringASTNode alloc] init];
-    [self parseCommonKeywordsFromSchema:stringSchema intoNode:node];
-
     node.maximumLength = [self parseUnsignedIntegerForKey:TWTJSONSchemaKeywordMaxLength schema:stringSchema];
     node.minimumLength = [self parseUnsignedIntegerForKey:TWTJSONSchemaKeywordMinLength schema:stringSchema];
-    // Does not check validity of regular expression, because definition is: "MUST be a string. This string SHOULD be a valid Regular Expression."
     node.pattern = [self parseStringForKey:TWTJSONSchemaKeywordPattern schema:stringSchema];
-
-    return node;
+    if (node.pattern) {
+        NSError *error = nil;
+        NSRegularExpression *regularExpression = [NSRegularExpression regularExpressionWithPattern:node.pattern options:0 error:&error];
+        if (error) {
```

Comment:
Here, should I be checking if `error` is present or if `regularExpression` is nil? The documentation wasn't clear whether it would definitely return nil but did seem to guarantee the error.